### PR TITLE
[TD-32792] perf(taos-optin): optimizing tmq poll

### PR DESCRIPTION
**Pull request by qevolg**
Optimizing tmq poll

### CODE REVIEW

File: taos-optin/src/tmq/raw.rs [hunk 1]
```diff
@@ -332,18 +309,79 @@ pub(super) mod tmq {
...
+                                tracing::warn!("Send res failed, cache res: {:?}", res);
+                                cache = res;
+                            }
+                            tracing::debug!(elapsed = ?elapsed.elapsed(), "Use cache, poll next message");
+                            continue;
+                        }
+
+                        let ptr = ptr.0;
+                        tracing::debug!("Calling C function `tmq_consumer_poll` with ptr: {ptr:?}, timeout: {timeout}");
+                        let res = unsafe { (tmq.tmq_consumer_poll)(ptr, timeout) };
```
Comment (sheyanjie-qq) [inline taos-optin/src/tmq/raw.rs:343]: ptr is consumer?

File: taos-optin/src/tmq/raw.rs [hunk 2]
```diff
@@ -332,18 +309,79 @@ pub(super) mod tmq {
...
+                            if let Err(_) = sender.send(None) {
+                                tracing::warn!("Send res failed");
+                            }
+                            tracing::debug!(elapsed = ?elapsed.elapsed(), "Res is null, poll next message");
+                            continue;
+                        }
+
+                        let res = unsafe { RawRes::from_ptr_unchecked(c.clone(), res) };
+                        if let Err(res) = sender.send(Some(res)) {
+                            tracing::warn!("Send res failed, cache res: {res:?}");
```
Comment (sheyanjie-qq) [inline taos-optin/src/tmq/raw.rs:358]: warn -> debug

File: taos-optin/src/tmq/mod.rs [hunk 3]
```diff
@@ -2495,4 +2471,85 @@ mod async_tests {
...
+                    offset.vgroup_id()
+                );
+
+                if let Some(data) = message.into_data() {
+                    while let Some(block) = data.fetch_raw_block().await? {
+                        let table_name = block.table_name().unwrap();
+                        let records: Vec<Record> = block.deserialize().try_collect()?;
+                        debug!(
+                            "table_name: {}, got {} records: {:#?}",
+                            table_name,
```
Comment (sheyanjie-qq) [inline taos-optin/src/tmq/mod.rs:2540]: check data

**Review by sheyanjie-qq — APPROVED** [review]

File: taos-optin/src/tmq/raw.rs [hunk 4]
```diff
@@ -329,21 +313,82 @@ pub(super) mod tmq {
...
+            let safe_tmq = SafeTmqT(self.as_ptr());
+            let tmq_api = self.tmq_api;
+            let api = self.api.clone();
+            let timeout = self.timeout;
+
+            std::thread::spawn(move || {
+                futures::executor::block_on(async {
+                    let safe_tmq = safe_tmq;
+                    let mut cache: Option<RawRes> = None;
+                    while let Some(sender) = receiver.recv().await {
```
Comment (zitsen) [inline taos-optin/src/tmq/raw.rs:337]: Only receiver.recv() is async method here, please use blocking method instead, such as flume::Receiver::recv()

**Review by zitsen — CHANGES_REQUESTED** [review]

**Review by sheyanjie-qq — APPROVED** [review]

**Review by zitsen — APPROVED** [review]

**Pull request merged**